joe/joe.py

```diff
@@ -40,18 +41,18 @@ def _get_data_dir(path):
 
 def _walk_gitignores():
     '''Recurse over the data directory and return all .gitignore file names'''
-    l = []
-    for root, subFolders, files in os.walk(DATA_DIR):
-        l += [f.replace('.gitignore', '') \
-                for f in files if f.endswith('.gitignore')]
-    return sorted(l)
+    gitignore = []
+    for _, _, files in os.walk(DATA_DIR):
+        gitignore += [f.replace('.gitignore', '')
+                      for f in files if f.endswith('.gitignore')]
+    return sorted(gitignore)
 
 
 # Where all gitignore files are
 DATA_DIR = _get_data_dir('*.gitignore')
 # Load up names for all gitignore files
 GITIGNORE_RAW = _walk_gitignores()
-GITIGNORE = [f.lower() for f in GITIGNORE_RAW]
+GITIGNORE = [filename.lower() for filename in GITIGNORE_RAW]
```

Comment:
Why go out of your way to obfuscate code? Readability is important. The comment above is too necessary for understanding what's happening in the remaining two lines.